htlcswitch/link.go

```diff
@@ -1008,16 +1025,26 @@ out:
 			}
 
 			// We'll check to see if we should update the fee rate
-			// based on our current set fee rate.
-			commitFee := l.channel.CommitFeeRate()
-			if !shouldAdjustCommitFee(feePerKw, commitFee) {
+			// based on our current set fee rate and our max fee
+			// allocation. Our max fee allocation depends on the
+			// configurable percentage and our available balance at
+			// the time, which is composed of our spendable balance
+			// and the fees currently committed to the commitment
+			// transaction.
+			commitFeeRate := l.channel.CommitFeeRate()
+			balance := l.channel.AvailableBalance().ToSatoshis() +
+				lnwallet.CalcFee(commitFeeRate)
```

Comment:
I think most of this logic can be lifted in to `shouldAdjustCommitFe` (first step may be to make it a method).

Reply:
I opted not to do so initially because of the additional test changes that it would require, though I can update the PR if desired.